api/v1/services/CheckInService.js

```diff
@@ -69,42 +69,39 @@ module.exports.createCheckIn = function (attributes){
+                              var message = "There are no remaining unassigned network credentials";
+                              var source = "NetworkCredential";
+                              throw new errors.UnprocessableRequestError(message, source);
+                          }
+                          var updates = {
+                              "userId": attributes.userId,
+                              "assigned": true
+                          };
+                          networkCredential.set(updates, {patch:true});
+                          return networkCredential.save(null, {transacting: t})
+                          .then(function(creds){
+                              return {
+                                  "checkin": model,
+                                  "credentials": creds
+                              };
+                          });
+                      });
+                  } else {
+                      return {"checkin": model, "credentials": null};
+                  }
+              });
+          });
         })
-        .then(function (existingCheckin) {
-            if (!_.isNull(existingCheckin)) {
-                var message = "A check in record already exists for this user";
-                var source = "userId";
-                throw new errors.InvalidParameterError(message, source);
-            }
-            return CheckIn.transaction(function (t) {
-                return checkin.save(null, {transacting: t})
-                .then(function(model){
-                    if(credentialsRequested){
-                        return NetworkCredential.findUnassigned()
-                        .then(function(networkCredential){
-                            if (_.isNull(networkCredential)) {
-                                var message = "There are no remaining unassigned network credentials";
-                                var source = "NetworkCredential";
-                                throw new errors.UnprocessableRequestError(message, source);
-                            }
-                            var updates = {
-                                "userId": attributes.userId,
-                                "assigned": true
-                            };
-                            networkCredential.set(updates, {patch:true});
-                            return networkCredential.save(null, {transacting: t})
-                            .then(function(creds){
-                                return {
-                                    "checkin": model,
-                                    "credentials": creds
-                                };
-                            });
-                        });
-                    } else {
-                        return {"checkin": model, "credentials": null};
-                    }
-                });
-            });
-        });
+        .catch((err) => err.code === errors.Constants.DupEntry, function (err) {
```

Comment:
As far as I understand Bluebird will let you do `.catch(ErrorType, handlerFunc)` or `.catch(predicateFunc, handlerFunc)` which will pass the caught `err`'s on to the handlerFunc but I don't know how one would pass in a `msg`, `src` parameter as well